plugins/feeds/public/badip_apache_ddos_ip.py

```diff
@@ -0,0 +1,47 @@
+import logging
+from datetime import timedelta
+
+from core.observables import Ip
+from core.feed import Feed
+from core.errors import ObservableValidationError
+
+# Time ranges for BadIp can be specified by y=year, day=d, minute=m, month=M
+class BadIPApacheDDOSIP(Feed):
+    default_values = {
+        'frequency': timedelta(hours=1),
+        'source': 'https://www.badips.com/get/list/apacheddos/0?age=1d',
+        'name': 'BadIPApacheDDOSIP',
+        'description': 'Badips.com is a community based IP blacklist service. This list is known hosts performing DDoS against Web Servers.'
+    }
+
+    def update(self):
+        # First run will backfill data going back 10 years.
+        # Since you can declare date ranges this opens the opportunity to create a first_seen function based on time subtraction
+        if self.last_run is None:
+            self.source = "https://www.badips.com/get/list/apacheddos/0?age=10y"
+
+        for line in self.update_lines():
+            self.analyze(line)
+
+    def analyze(self, line):
+        if line.startswith('#'):
+            return
+
+        try:
+            line = line.strip()
+            parts = line.split()
```

Comment:
The source data seems to have one IP per line with no spaces (`split()` splits a string using spaces when no argument is provided), so do we really need to call `split()` and use `parts[0]` below? Same comment goes for most of the feeds below.